tests/v2/validation/Jenkinsfile_support_matrix_os

```diff
@@ -0,0 +1,234 @@
+                  // install yq
+                  sh returnStdout: true, script: 'wget -qO ./yq https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64'
+                  sh returnStdout: true, script: 'chmod a+x ./yq'
+                  def amis = env.RANCHER_AMI_LIST
+                  amiList = amis.split(',').each { obj -> 
+                  String ami
+                  String sshUser
+                  if (obj.contains(':')){
+                    def data = obj.split(':')
+                    ami = data[0]
+                    sshUser = data[1]
+                  } else {
+                    ami = obj
+                    sshUser = 'ec2-user'
+                  }
+
+                  // RKE1 Node Driver: awsNodeTemplate
+
+                  def awsNodeTemplateFilename = "awsNodeTemplate.yaml"
+                  def awsNodeTemplate = env.RANCHER_RKE1_NODE_CONFIG
+
+                  writeFile file: awsNodeTemplateFilename, text: awsNodeTemplate
+
+                  sh "./yq e '.awsNodeTemplate.ami = \"$ami\"' -i awsNodeTemplate.yaml"
+
+                  sh "./yq e '.awsNodeTemplate.sshUser = \"$sshUser\"' -i awsNodeTemplate.yaml"
+
+                  def awsNodeTemplateParam = readFile(file: './awsNodeTemplate.yaml')
+
+                  echo awsNodeTemplateParam
+
+                  // RKE2 Node Driver: awsMachineConfig
+
+                  def awsMachineConfigFilename = "awsMachineConfig.yaml"
+                  def awsMachineConfig = env.RANCHER_RKE2_NODE_CONFIG
+
+                  writeFile file: awsMachineConfigFilename, text: awsMachineConfig
+
+                  sh "./yq e '.awsMachineConfig.ami = \"$ami\"' -i awsMachineConfig.yaml"
+
+                  sh "./yq e '.awsMachineConfig.sshUser = \"$sshUser\"' -i awsMachineConfig.yaml"
+
+                  def awsMachineConfigParam = readFile(file: './awsMachineConfig.yaml')
+
+                  echo awsMachineConfigParam
+
+                  // Single custom cluster config: awsEC2Config
+
+                  def awsEC2ConfigFilename = "awsEC2Config.yaml"
+                  def awsEC2Config = env.RANCHER_CUSTOM_CONFIG
+
+                  writeFile file: awsEC2ConfigFilename, text: awsEC2Config 
+
+                  sh "./yq e '.awsEC2Configs.awsEC2Config.[0].awsAMI = \"$ami\"' -i awsEC2Config.yaml"
+
+                  sh "./yq e '.awsEC2Configs.awsEC2Config.[0].awsUser = \"$sshUser\"' -i awsEC2Config.yaml"
+
+                  def awsEC2ConfigParam = readFile(file: './awsEC2Config.yaml')
+
+                  echo awsEC2ConfigParam
+
+                  // // RKE1 Custom Cluster: awsEC2Config
+
+                  // def rke1CustomFilename = "rke1CustomConfig.yaml"
+                  // def rke1CustomConfig = env.RANCHER_RKE1_CUSTOM_CONFIG
+
+                  // writeFile file: rke1CustomFilename, text: rke1CustomConfig
+
+                  // sh "./yq e '.awsEC2Configs.awsEC2Config.[0].awsAMI = \"$ami\"' -i rke1CustomConfig.yaml"
+
```

Comment:
looks great can you clean this up and then its good to go

Reply:
took out the comments & unnecessary code